crates/object_detection/src/pose_interpretation.rs

```diff
@@ -0,0 +1,256 @@
+                } else {
+                    None
+                }
+            })
+            .min_by_key(|(_, distance)| NotNan::new(*distance).unwrap());
+
+        match pose_type_tuple {
+            Some((pose, distance_to_referee_position))
+                if distance_to_referee_position < distance_to_referee_position_threshold =>
+            {
+                Some(*pose)
+            }
+            _ => None,
+        }
+    }
+
+    pub fn interpret_pose(
+        human_pose: &Option<HumanPose>,
+        keypoint_confidence_threshold: f32,
+        shoulder_angle_threshold: f32,
+    ) -> PoseType {
+        match human_pose {
+            Some(pose)
+                if Self::is_over_head_arms(
+                    pose.keypoints,
+                    keypoint_confidence_threshold,
+                    shoulder_angle_threshold,
+                ) =>
+            {
+                PoseType::OverheadArms
+            }
+            _ => PoseType::default(),
+        }
+    }
+
+    pub fn is_over_head_arms(
+        keypoints: Keypoints,
+        keypoint_confidence_threshold: f32,
+        shoulder_angle_threshold: f32,
+    ) -> bool {
+        struct RotatedPixel;
+
+        let are_hands_visible = keypoints.left_hand.confidence > keypoint_confidence_threshold
+            && keypoints.right_hand.confidence > keypoint_confidence_threshold;
+        let are_hands_over_shoulder = keypoints.left_shoulder.point.y()
+            > keypoints.left_hand.point.y()
+            && keypoints.right_shoulder.point.y() > keypoints.right_hand.point.y();
+
+        let left_to_right_shoulder =
+            keypoints.right_shoulder.point.coords() - keypoints.left_shoulder.point.coords();
+        let shoulder_line_angle =
+            f32::atan2(left_to_right_shoulder.y(), left_to_right_shoulder.x());
+        let shoulder_rotation =
+            Transform::<Pixel, RotatedPixel, nalgebra::Isometry2<_>>::rotation(shoulder_line_angle);
+
+        let left_shoulder = shoulder_rotation * keypoints.left_shoulder.point;
+        let right_shoulder = shoulder_rotation * keypoints.right_shoulder.point;
+        let left_elbow = shoulder_rotation * keypoints.left_elbow.point;
+        let right_elbow = shoulder_rotation * keypoints.right_elbow.point;
+        let left_shoulder_to_elbow = left_elbow.coords() - left_shoulder.coords();
+        let right_shoulder_to_elbow = right_elbow.coords() - right_shoulder.coords();
+
+        let is_left_shoulder_angled_up =
+            f32::atan2(right_shoulder_to_elbow.y(), right_shoulder_to_elbow.x())
+                > shoulder_angle_threshold;
+        let is_right_shoulder_angled_up =
+            f32::atan2(left_shoulder_to_elbow.y(), -left_shoulder_to_elbow.x())
+                > shoulder_angle_threshold;
+
+        if are_hands_visible {
+            are_hands_over_shoulder
+        } else {
+            is_right_shoulder_angled_up && is_left_shoulder_angled_up
+        }
```

Comment:
This looks very complicated, takes me a very long time to parse what is going on here

Reply:
Done